core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2567,7 +2585,10 @@ public boolean doCheckIsUserInRole(String userName, String roleName) throws User
                                 memberOfProperty);
             } else {
                 // use cache
-                LdapName ldn = (LdapName)userCache.get(userName);
+                LdapName ldn = null;
+                if (userName != null) {
+                    ldn = (LdapName) userCache.get(userName);
```

Comment:
It may be good to validate username as an input at the begining of the function and then perform further checking, WDYT?